ovos_tts_plugin_piper/__init__.py

```diff
@@ -123,60 +123,112 @@ class PiperTTSPlugin(TTS):
-        if isinstance(model, list):
-            model = model[0]
-
-        if not os.path.isfile(model) and model in self.voice2url:
-            xdg_p = f"{xdg_data_home()}/piper_tts/{model}"
-            if not os.path.isdir(xdg_p):
-                model = self.voice2url[model]
-
-        if model.startswith("http"):
-            m = model.split("/")[-1]
-            xdg_p = f"{xdg_data_home()}/piper_tts/{m.split('.')[0]}"
-
-            model_tar = f"{xdg_p}/{m}"
-            if not os.path.isfile(model_tar):
-                LOG.info(f"downloading piper model: {model}")
-                os.makedirs(xdg_p, exist_ok=True)
-                data = requests.get(model)
-                with open(model_tar, "wb") as f:
-                    f.write(data.content)
-
-                with tarfile.open(model_tar) as file:
-                    file.extractall(xdg_p)
-            for f in os.listdir(xdg_p):
-                if f.endswith(".onnx"):
-                    model = f"{xdg_p}/{f}"
-                    LOG.debug(f"selected model: {model}")
-                    break
+        if self.voice == "default":
+            if self.lang.startswith("en"):
+                # alan pope is the default english voice of mycroft/OVOS
+                self.voice = "alan-low"
             else:
-                raise FileNotFoundError("onnx model not found")
+                self.voice = self.lang2voices.get(lang) or \
+                             self.lang2voices.get(lang.split("-"[0])) or \
+                             "alan-low"
 
-        self.model = model
-        self.model_json = self.config.get("model_json") or self.model + ".json"
         self.use_cuda = self.config.get("use_cuda", False)
-        self.engine = Piper(self.model, config_path=self.model_json, use_cuda=self.use_cuda)
-
-        self.speaker = self.config.get("speaker")  # "Id of speaker (default: 0)"
         self.noise_scale = self.config.get("noise-scale")  # generator noise
         self.length_scale = self.config.get("length-scale")  # Phoneme length
         self.noise_w = self.config.get("noise-w")  # Phoneme width noise
-        self.g2p = PiperG2P(self.engine)
 
-    def get_tts(self, sentence, wav_file, lang=None, speaker=None):
+        # pre-load models
+        preload_voices = self.config.get("preload_voices") or []
+        preload_langs = self.config.get("preload_langs") or [self.lang]
+
+        for lang in preload_langs:
+            if lang not in self.lang2voices:
+                lang = lang.split("-")[0]
+            voice = self.lang2voices.get(lang)
+            if voice and voice not in preload_voices:
+                preload_voices.append(voice)
+
+        for voice in preload_voices:
+            self.get_model(voice=voice)
+
+    def get_model(self, lang=None, voice=None, speaker=None):
+
+        # find default voice  (should be called model not voice....)
+        if voice is None and lang is not None:
+            if lang.startswith("en"):
```

Comment:
Same here. Can this have `and "voice" == <None/default>` so that another English voice can be used

Reply:
same as above, this is only when voice is not requested at all